homeassistant/components/lamarzocco/switch.py

```diff
@@ -0,0 +1,193 @@
+"""Switch platform for La Marzocco espresso machines."""
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from typing import Any
+
+from lmcloud.const import LaMarzoccoModel
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import LaMarzoccoEntity, LaMarzoccoEntityDescription
+from .lm_client import LaMarzoccoClient
+
+ATTR_MAP_MAIN_GS3_AV = [
+    "dose_k1",
+    "dose_k2",
+    "dose_k3",
+    "dose_k4",
+    "dose_k5",
+]
+
+ATTR_MAP_AUTO_ON_OFF = [
```

Comment:
Why are these all extra state attributes and not own entities? We have time entities or maybe even better calendar entities

Reply:
problem with a lot of sensor entities is you can't really group them and it just gets messy. Thought about using a calendar, but then you'd set it for one week and it would be reflected for every week, which is also not great imo